doc/jms.adoc

```diff
@@ -0,0 +1,315 @@
+
+You can use SmallRye Reactive Messaging to receive and send JMS messages.
+
+=== Dependency
+
+To enable the JMS support, you need the following dependency:
+
+[source,xml,subs=attributes+]
+----
+<dependency>
+  <groupId>io.smallrye.reactive</groupId>
+  <artifactId>smallrye-reactive-messaging-jms</artifactId>
+  <version>{version}</version>
+</dependency>
+----
+
+The connector is named `smallrye-jms`.
+
+=== Connection Factory Requirement
+
+To interact with JMS, you need to _expose_ a `javax.jms.ConnectionFactory` as CDI bean.
+For example, you need a bean producing the `ConnectionFactory`:
+
+[source, java]
+----
+@ApplicationScoped
+public class ConnectionFactoryBean {
+
+    @Inject @ConfigProperty(name="jms-username")
+    String username;
+
+    @Inject @ConfigProperty(name="jms-password")
+    String password;
+
+    @Produces
+    ConnectionFactory factory() {
+        ActiveMQJMSConnectionFactory factory = new ActiveMQJMSConnectionFactory(
+                "tcp://localhost:61616",
+                username, password);
+        return factory;
+    }
+
+}
+----
+
+The JMS connector is consuming the produced _factory_ to connect to the broker.
+
+==== Selecting ConnectionFactory
+
+In case you have several connection factories, you can use the `@Named` qualified on your factory to specify the name.
+Then, in the channel configuration, configure the name as follows:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.connection-factory-name=my-factory-name
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.connection-factory-name=my-factory-name
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.connection-factory-name=my-factory-name
+----
+
+==== Configure credentials
+
+To configure the _username_ and _password_ to connect to the broker and create a session, use the following properties:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.username=my-username
+mp.messaging.connector.smallrye-jms.password=secret
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.username=my-username
+mp.messaging.incoming.my-channel.password=secret
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.username=my-username
+mp.messaging.outgoing.my-channel.password=secret
+----
+
+==== Session mode
+
+By default, the connector uses the `AUTO_ACKNOWLEDGE` session mode.
+You can configure it using:
+
+[source,properties]
+----
+# Configure the connector globally
+mp.messaging.connector.smallrye-jms.session-mode=AUTO_ACKNOWLEDGE
+# Configure a specific incoming channel
+mp.messaging.incoming.my-channel.session-mode=AUTO_ACKNOWLEDGE
+# Configure a specific outgoing channel
+mp.messaging.outgoing.my-channel.session-mode=AUTO_ACKNOWLEDGE
+----
+
+Accepted values are:
+
+* `AUTO_ACKNOWLEDGE`
+* `SESSION_TRANSACTED`
+* `CLIENT_ACKNOWLEDGE`
+* `DUPS_OK_ACKNOWLEDGE`
```

Comment:
The spec needs to state an `IllegalArgumentException` occurs if the value provided is not among the accepted values.

Reply:
By spec, do you mean the documentation? 

Also, this would result in a DeploymentException (caused by this IllegalArgumentException). 
I'm going to add a sentence in the documentation and a test (because we never know...)

Reply:
Done.